bindings/aws/s3/s3.go

```diff
@@ -79,10 +80,13 @@ func (s *AWSS3) Init(metadata bindings.Metadata) error {
 	if err != nil {
 		return err
 	}
+
+	cfg := aws.NewConfig().WithS3ForcePathStyle(m.ForcePathStyle)
+
 	s.metadata = m
-	s.s3Client = s3.New(session)
-	s.downloader = s3manager.NewDownloader(session)
-	s.uploader = s3manager.NewUploader(session)
+	s.s3Client = s3.New(session, cfg)
+	s.downloader = s3manager.NewDownloaderWithClient(s.s3Client)
+	s.uploader = s3manager.NewUploaderWithClient(s.s3Client)
```

Comment:
is there a reason to change to `NewDownloaderWithClient` from `NewDownloader` ?

Reply:
`S3ForcePathStyle` only can be set with s3 client or session. I couldn't find a way to set it up in `NewDownloader`.

Reply:
@mukundansundar

Reply:
> `S3ForcePathStyle` only can be set with s3 client or session. I couldn't find a way to set it up in `NewDownloader`.

@rainfd if it can be set in the session why not set it there, since session is already being used?

Reply:
> @rainfd if it can be set in the session why not set it there, since session is already being used?

`func GetClient(accessKey string, secretKey string, sessionToken string, region string, endpoint string) (*session.Session, error)`
session get by `GetClient` in authentication/aws/aws.go,  setting ForcePathStyle here is also ok, like
`func GetClient(accessKey string, secretKey string, sessionToken string, region string, endpoint string, forcePathStyle bool) (*session.Session, error) {` 
 but `GetClient` is used by other aws componment. Should I add modify here?

Reply:
Thanks fir explaining that ... And no no need to make change in the common method. 
Can you add a docs PR for this and also unit tests to make sure that the metadata is parsed correctly?

Reply:
@mukundansundar test and docs have already added.
dapr/docs#2054